src/OgAccess.php

```diff
@@ -262,36 +262,37 @@ public function userAccessEntity($operation, EntityInterface $entity, AccountInt
     }
 
     $is_group_content = $this->groupAudienceHelper->hasGroupAudienceField($entity_type_id, $bundle);
-    $cache_tags = $entity_type->getListCacheTags();
-
-    // The entity might be a user or a non-user entity.
-    $groups = $entity->getEntityTypeId() == 'user' ? $this->membershipManager->getUserGroups($entity) : $this->membershipManager->getGroups($entity);
-
-    if ($is_group_content && $groups) {
-      $forbidden = AccessResult::forbidden()->addCacheTags($cache_tags);
-      foreach ($groups as $entity_groups) {
-        foreach ($entity_groups as $group) {
-          // Check if the operation matches a group content entity operation
-          // such as 'create article content'.
-          $operation_access = $this->userAccessGroupContentEntityOperation($operation, $group, $entity, $user);
-
-          if ($operation_access->isAllowed()) {
-            return $operation_access->addCacheTags($cache_tags);
-          }
-
-          // Check if the operation matches a group level operation such as
-          // 'subscribe without approval'.
-          $user_access = $this->userAccess($group, $operation, $user);
-          if ($user_access->isAllowed()) {
-            return $user_access->addCacheTags($cache_tags);
+    if ($is_group_content) {
+      $cache_tags = $entity_type->getListCacheTags();
+
+      // The entity might be a user or a non-user entity.
+      $groups = $entity->getEntityTypeId() == 'user' ? $this->membershipManager->getUserGroups($entity) : $this->membershipManager->getGroups($entity);
```

Comment:
Or we could test for Drupal\user\UserInterface?

Also might have been useful to have this check within getGroups().

Reply:
I agree with this proposal but these lines of code are not in fact being changed here, this PR just introduces this single line and moves the related code in it:

```
if ($is_group_content) {
  // All code in here is in fact unchanged, but shows up in Github as a huge chunk of changed code.
}
```